edns.go

```diff
@@ -225,77 +230,90 @@ type EDNS0_SUBNET struct {
 // Option implements the EDNS0 interface.
 func (e *EDNS0_SUBNET) Option() uint16 { return EDNS0SUBNET }
 
-func (e *EDNS0_SUBNET) pack() ([]byte, error) {
-	b := make([]byte, 4)
+func (e *EDNS0_SUBNET) Len() int {
+	l := 4
+	switch e.Family {
```

Comment:
think this can be more easily done with an if
`if e.Family == 1 { return 4 }` and then handle the `Family == 2 ` case. Note sure what the RFC says about other families.

Reply:
I reviewed the RFC. It says

```
   The format of the address part depends on the value of FAMILY.  This
   document only defines the format for FAMILY 1 (IPv4) and FAMILY 2
   (IPv6), which are as follows:
```

Sounds to me like FAMILY only controls the format of ADDRESS, but SOURCE PREFIX-LENGTH and SCOPE PREFIX-LENGTH retain their meaning. In any case, only IPv4 and IPv6 are defined, to which this generalisation holds.

Therefore, I simplified the code to
````
return 4 + int((e.SourceNetmask + 8 - 1) / 8)
```